common/request.go

```diff
@@ -441,3 +454,49 @@ func (r *NormalizedRequest) Validate() error {
 
 	return nil
 }
+
+// IsCompositeRequest returns whether this request is a top-level composite request
+func (r *NormalizedRequest) IsCompositeRequest() bool {
+	if r == nil {
+		return false
+	}
+	return r.isCompositeRequest.Load()
+}
+
+func (r *NormalizedRequest) SetIsCompositeRequest(isComposite bool) {
+	if r == nil {
+		return
+	}
+	r.isCompositeRequest.Store(isComposite)
+}
```

Comment:
Remove the `isCompositeRequest` and setter.

Instead update the IsCompositeRequest to equality check against TypeNone...